packages/driver/cypress/integration/e2e/multi-domain/multi_domain_yield_spec.ts

```diff
@@ -0,0 +1,253 @@
+
+      done()
+    })
+
+    cy.switchToDomain('foobar.com', () => {
+      cy
+      .get('[data-cy="dom-check"]')
+    }).invoke('text')
+    .should('equal', 'From a secondary domain')
+  })
+
+  it('throws if an object contains a function', (done) => {
+    cy.on('fail', (err) => {
+      assertLogLength(logs, 7)
+      expect(logs[6].get('error')).to.eq(err)
+      expect(err.message).to.include('`cy.switchToDomain()` could not serialize the subject due to one of its properties not being supported by the structured clone algorithm.')
+
+      done()
+    })
+
+    cy.switchToDomain('foobar.com', () => {
+      cy.wrap({
+        key: () => {
+          return 'whoops'
+        },
+      })
+    }).invoke('key').should('equal', 'whoops')
+  })
+
+  it('throws if an object contains a symbol', (done) => {
+    cy.on('fail', (err) => {
+      assertLogLength(logs, 7)
+      expect(logs[6].get('error')).to.eq(err)
+      expect(err.message).to.include('`cy.switchToDomain()` could not serialize the subject due to one of its properties not being supported by the structured clone algorithm.')
+
+      done()
+    })
+
+    cy.switchToDomain('foobar.com', () => {
+      cy.wrap({
+        key: Symbol('whoops'),
+      })
+    }).should('equal', undefined)
+  })
+
+  it('throws if an object is a function', (done) => {
+    cy.on('fail', (err) => {
+      assertLogLength(logs, 7)
+      expect(logs[6].get('error')).to.eq(err)
+      expect(err.message).to.include('`cy.switchToDomain()` could not serialize the subject due to functions not being supported by the structured clone algorithm.')
+
+      done()
+    })
+
+    cy.switchToDomain('foobar.com', () => {
+      cy.wrap(() => {
+        return 'text'
+      })
+    }).then((obj) => {
+      // @ts-ignore
+      obj()
+    })
+  })
+
+  it('throws if an object is a symbol', (done) => {
+    cy.on('fail', (err) => {
+      assertLogLength(logs, 7)
+      expect(logs[6].get('error')).to.eq(err)
+      expect(err.message).to.include('`cy.switchToDomain()` could not serialize the subject due to symbols not being supported by the structured clone algorithm.')
+
+      done()
+    })
+
+    cy.switchToDomain('foobar.com', () => {
+      cy.wrap(Symbol('symbol'))
+    }).should('equal', 'symbol')
+  })
+
+  // NOTE: Errors can only be serialized on chromium browsers.
+  it('yields an error if an object contains an error', (done) => {
+    const isChromium = Cypress.isBrowser({ family: 'chromium' })
+
+    cy.on('fail', (err) => {
+      if (!isChromium) {
+        assertLogLength(logs, 7)
+        expect(logs[6].get('error')).to.eq(err)
+        expect(err.message).to.include('`cy.switchToDomain()` could not serialize the subject due to one of its properties not being supported by the structured clone algorithm.')
+      }
+
+      done()
+    })
+
+    cy.switchToDomain('foobar.com', () => {
+      cy.wrap({
+        key: new Error('Boom goes the dynamite'),
```

Comment:
Out of scope here, but I am also wondering if we should do something similar with test errors. In other words, only try to preprocess the error if the browser can't serialize it through `postMessage`. Otherwise, just send it as is as opposed to always preprocessing the error.